pkg/api/openshift.go

```diff
@@ -0,0 +1,55 @@
+package api
+
+import (
+	"sync"
+
+	authv1 "github.com/openshift/client-go/authorization/clientset/versioned/typed/authorization/v1"
+	routev1 "github.com/openshift/client-go/route/clientset/versioned/typed/route/v1"
+	"github.com/pkg/errors"
+	log "github.com/sirupsen/logrus"
+	"k8s.io/client-go/rest"
+)
+
+// OpenshiftClient - Client to interact with openshift api
+type OpenshiftClient struct {
+	authClient  authv1.AuthorizationV1Interface
+	routeClient routev1.RouteV1Interface
+}
+
+var instances struct {
+	Openshift *OpenshiftClient
+}
+
+var once struct {
+	Openshift sync.Once
+}
+
+// Openshift - Create a new openshift client if needed, returns reference
+func Openshift(config *rest.Config) (*OpenshiftClient, error) {
+	errMsg := "Something went wrong while initializing openshift client!\n"
+	once.Openshift.Do(func() {
+		client, err := newOpenshift(config)
+		if err != nil {
+			log.Fatal(errMsg)
```

Comment:
Shouldn't it return on error?

Reply:
Yes, you're right that's residual code from previous work.

Reply:
Yes, it will. Thanks

Reply:
@alexander-demichev, actually if Openshift creation fails it's fatal.

Reply:
Hmm, shouldn't we skip report of API resources in this case? If not, then we also need to panic in k8s client initialization.

Reply:
Yeah I suppose there are 2 different approaches, whether we trigger panic or escalate.
I initially thought that not being able to create an o7t client was a blocker and so is k8s because the report does rely on the APIs. 
Let's just escalate for now, I created issue https://github.com/fusor/cpma/issues/272 
to allow us to move on as we need o7t.